File: lib/src/web_workers/shared/serialized_types.dart
// This file contains interface versions of browser types that can be serialized to Plain Old

// JavaScript Objects
library angular2.src.web_workers.shared.serialized_types;

class LocationType {
  String href;
  String protocol;
  String host;
  String hostname;
  String port;
  String pathname;
  String search;
  String hash;
  String origin;
  LocationType(this.href, this.protocol, this.host, this.hostname, this.port,
      this.pathname, this.search, this.hash, this.origin) {}
}

File: modules/@angular/platform-browser/src/web_workers/shared/serialized_types.ts
/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

// This file contains interface versions of browser types that can be serialized to Plain Old
// JavaScript Objects
export class LocationType {
  constructor(
      public href: string, public protocol: string, public host: string, public hostname: string,
      public port: string, public pathname: string, public search: string, public hash: string,
      public origin: string) {}
}
